pytest failure

Hi, I'm reviewing your software as part of your JOSS submission. I have everything installed but am seeing a failure in one of your tests.

```
========================================================= test session starts ==========================================================
platform linux -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/opc/MSMetaEnhancer
plugins: aiohttp-0.3.0
collected 39 items                                                                                                                     

tests/test_BridgeDB.py ...                                                                                                       [  7%]
tests/test_CIR.py ...                                                                                                            [ 15%]
tests/test_CTS.py ....                                                                                                           [ 25%]
tests/test_IDSM.py X..                                                                                                           [ 33%]
tests/test_NLM.py ...                                                                                                            [ 41%]
tests/test_PubChem.py .....                                                                                                      [ 53%]
tests/test_annotator.py ....                                                                                                     [ 64%]
tests/test_application.py ..                                                                                                     [ 69%]
tests/test_converter.py ........                                                                                                 [ 89%]
tests/test_curator.py .F..                                                                                                       [100%]

=============================================================== FAILURES ===============================================================
____________________________________ test_filter_invalid_metadata[metadata0-validated_metadata0-1] _____________________________________

metadata = {'formula': 'CH4', 'inchi': 'InChI=1S/CH4/h1H4', 'iupac_name': 'methane', 'smiles': 'C'}
validated_metadata = {'formula': 'CH4', 'inchi': 'InChI=1S/CH4/h1H4', 'iupac_name': 'methane'}, logs_size = 1

    @pytest.mark.parametrize('metadata, validated_metadata, logs_size', [
        [{'formula': 'CH4', 'smiles': 'C', 'iupac_name': 'methane', 'inchi': 'InChI=1S/CH4/h1H4'},
         {'formula': 'CH4', 'iupac_name': 'methane', 'inchi': 'InChI=1S/CH4/h1H4'}, 1],
        [{'inchikey': '<html>random content</html>'}, {}, 1],
        [{'smiles': 'CC(NC(C)=O)C#N'}, {'smiles': 'CC(NC(C)=O)C#N'}, 0]
    ])
    def test_filter_invalid_metadata(metadata, validated_metadata, logs_size):
        warning = LogRecord(dict())
        job = Job(('smiles', 'inchi', 'converter'))
        curator = Curator()
>       assert curator.filter_invalid_metadata(metadata, warning, job) == validated_metadata
E       AssertionError: assert {'formula': '...'smiles': 'C'} == {'formula': '...e': 'methane'}
E         Omitting 3 identical items, use -vv to show
E         Left contains 1 more item:
E         {'smiles': 'C'}
E         Use -v to get more diff

tests/test_curator.py:24: AssertionError
=========================================================== warnings summary ===========================================================
../joss_mamba/envs/MSMetaEnhancer/lib/python3.8/site-packages/pyteomics/auxiliary/patch.py:12
../joss_mamba/envs/MSMetaEnhancer/lib/python3.8/site-packages/pyteomics/auxiliary/patch.py:12
  /home/opc/joss_mamba/envs/MSMetaEnhancer/lib/python3.8/site-packages/pyteomics/auxiliary/patch.py:12: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if LooseVersion(pv) < LooseVersion('0.17'):

tests/test_BridgeDB.py:11
  /home/opc/MSMetaEnhancer/tests/test_BridgeDB.py:11: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency()

tests/test_BridgeDB.py:16
  /home/opc/MSMetaEnhancer/tests/test_BridgeDB.py:16: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency(depends=["test_service_available"])

tests/test_CIR.py:8
  /home/opc/MSMetaEnhancer/tests/test_CIR.py:8: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency()

tests/test_CIR.py:13
  /home/opc/MSMetaEnhancer/tests/test_CIR.py:13: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency(depends=["test_service_available"])

tests/test_CTS.py:9
  /home/opc/MSMetaEnhancer/tests/test_CTS.py:9: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency()

tests/test_CTS.py:14
  /home/opc/MSMetaEnhancer/tests/test_CTS.py:14: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency(depends=["test_service_available"])

tests/test_IDSM.py:15
  /home/opc/MSMetaEnhancer/tests/test_IDSM.py:15: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency()

tests/test_IDSM.py:20
  /home/opc/MSMetaEnhancer/tests/test_IDSM.py:20: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency(depends=["test_service_available"])

tests/test_NLM.py:10
  /home/opc/MSMetaEnhancer/tests/test_NLM.py:10: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency()

tests/test_NLM.py:15
  /home/opc/MSMetaEnhancer/tests/test_NLM.py:15: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency(depends=["test_service_available"])

tests/test_PubChem.py:14
  /home/opc/MSMetaEnhancer/tests/test_PubChem.py:14: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency()

tests/test_PubChem.py:19
  /home/opc/MSMetaEnhancer/tests/test_PubChem.py:19: PytestUnknownMarkWarning: Unknown pytest.mark.dependency - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
    @pytest.mark.dependency(depends=["test_service_available"])

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================= short test summary info ========================================================
FAILED tests/test_curator.py::test_filter_invalid_metadata[metadata0-validated_metadata0-1] - AssertionError: assert {'formula': '......
======================================== 1 failed, 37 passed, 1 xpassed, 14 warnings in 29.43s =========================================
```

Using `-vv` shows

```
$ pytest -vv tests/test_curator.py 
========================================================= test session starts ==========================================================
platform linux -- Python 3.8.13, pytest-7.1.2, pluggy-1.0.0 -- /home/opc/joss_mamba/envs/MSMetaEnhancer/bin/python
cachedir: .pytest_cache
rootdir: /home/opc/MSMetaEnhancer
plugins: aiohttp-0.3.0
collected 4 items                                                                                                                      

tests/test_curator.py::test_fix_cas_number PASSED                                                                                [ 25%]
tests/test_curator.py::test_filter_invalid_metadata[metadata0-validated_metadata0-1] FAILED                                      [ 50%]
tests/test_curator.py::test_filter_invalid_metadata[metadata1-validated_metadata1-1] PASSED                                      [ 75%]
tests/test_curator.py::test_filter_invalid_metadata[metadata2-validated_metadata2-0] PASSED                                      [100%]

=============================================================== FAILURES ===============================================================
____________________________________ test_filter_invalid_metadata[metadata0-validated_metadata0-1] _____________________________________

metadata = {'formula': 'CH4', 'inchi': 'InChI=1S/CH4/h1H4', 'iupac_name': 'methane', 'smiles': 'C'}
validated_metadata = {'formula': 'CH4', 'inchi': 'InChI=1S/CH4/h1H4', 'iupac_name': 'methane'}, logs_size = 1

    @pytest.mark.parametrize('metadata, validated_metadata, logs_size', [
        [{'formula': 'CH4', 'smiles': 'C', 'iupac_name': 'methane', 'inchi': 'InChI=1S/CH4/h1H4'},
         {'formula': 'CH4', 'iupac_name': 'methane', 'inchi': 'InChI=1S/CH4/h1H4'}, 1],
        [{'inchikey': '<html>random content</html>'}, {}, 1],
        [{'smiles': 'CC(NC(C)=O)C#N'}, {'smiles': 'CC(NC(C)=O)C#N'}, 0]
    ])
    def test_filter_invalid_metadata(metadata, validated_metadata, logs_size):
        warning = LogRecord(dict())
        job = Job(('smiles', 'inchi', 'converter'))
        curator = Curator()
>       assert curator.filter_invalid_metadata(metadata, warning, job) == validated_metadata
E       AssertionError: assert {'formula': 'CH4',\n 'inchi': 'InChI=1S/CH4/h1H4',\n 'iupac_name': 'methane',\n 'smiles': 'C'} == {'formula': 'CH4', 'inchi': 'InChI=1S/CH4/h1H4', 'iupac_name': 'methane'}
E         Common items:
E         {'formula': 'CH4', 'inchi': 'InChI=1S/CH4/h1H4', 'iupac_name': 'methane'}
E         Left contains 1 more item:
E         {'smiles': 'C'}
E         Full diff:
E           {
E            'formula': 'CH4',
E            'inchi': 'InChI=1S/CH4/h1H4',
E            'iupac_name': 'methane',
E         +  'smiles': 'C',
E           }

tests/test_curator.py:24: AssertionError
=========================================================== warnings summary ===========================================================
../joss_mamba/envs/MSMetaEnhancer/lib/python3.8/site-packages/pyteomics/auxiliary/patch.py:12
../joss_mamba/envs/MSMetaEnhancer/lib/python3.8/site-packages/pyteomics/auxiliary/patch.py:12
  /home/opc/joss_mamba/envs/MSMetaEnhancer/lib/python3.8/site-packages/pyteomics/auxiliary/patch.py:12: DeprecationWarning: distutils Version classes are deprecated. Use packaging.version instead.
    if LooseVersion(pv) < LooseVersion('0.17'):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================= short test summary info ========================================================
FAILED tests/test_curator.py::test_filter_invalid_metadata[metadata0-validated_metadata0-1] - AssertionError: assert {'formula': 'CH4...
=============================================== 1 failed, 3 passed, 2 warnings in 1.20s ================================================
```

I am running on Linux, using Python 3.8.13 and installed MSMetaEnhancer using the instructions on your homepage.

--- a/tests/test_curator.py
+++ b/tests/test_curator.py
@@ -12,8 +12,6 @@ def test_fix_cas_number():
 
 
 @pytest.mark.parametrize('metadata, validated_metadata, logs_size', [
-    [{'formula': 'CH4', 'smiles': 'C', 'iupac_name': 'methane', 'inchi': 'InChI=1S/CH4/h1H4'},
-     {'formula': 'CH4', 'iupac_name': 'methane', 'inchi': 'InChI=1S/CH4/h1H4'}, 1],
     [{'inchikey': '<html>random content</html>'}, {}, 1],
     [{'smiles': 'CC(NC(C)=O)C#N'}, {'smiles': 'CC(NC(C)=O)C#N'}, 0]
 ])

